### PR TITLE
DataFrame:  split DataFrameHelper into MutableDataFrame and FieldCache

### PR DESCRIPTION
**Pull request by ryantxu**
While working with DataFrame, it would be nice to have an easier way to build them.

This PR does three key things:
* Moves all DataFrame manipulation/building into `MutableDataFrame`
* Moves DataFrame helper to `FieldCache` -- it is now mostly a way to find fields by type/name
* Adds a special function `getTimeField` for the most common case of finding the first time field.  No need to build a full FieldCache just to find the first time value and skip non-numbers

Its nice to get rid of a class name we never liked :)

### CODE REVIEW

File: packages/grafana-data/src/utils/dataFrameHelper.ts [hunk 1]
```diff
@@ -200,6 +200,10 @@ export class DataFrameHelper implements DataFrame {
   getFieldByName(name: string): Field | undefined {
     return this.fieldByName[name];
   }
+
+  toJSON() {
```
Comment (torkelo) [inline packages/grafana-data/src/utils/dataFrameHelper.ts]: toDTO?  

toJSON i would expect a json string returned

**Review by torkelo — COMMENTED** [review]
Nice, is MutableDataFrame more appropriate?

Should we remove the add methods on DataFrameHelper? 

What is the main difference?

File: packages/grafana-data/src/utils/dataFrameHelper.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-data/src/utils/dataFrameHelper.ts]: `toJSON` gets called by `JSON.stringify()` -- see:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON()_behavior

Comment (torkelo) [reply packages/grafana-data/src/utils/dataFrameHelper.ts]: ah, did not know that, then it makes perfect sense

**Review by torkelo — APPROVED** [review]

File: packages/grafana-data/src/utils/__snapshots__/csv.test.ts.snap [hunk 2]
```diff
@@ -30,7 +30,7 @@ Object {
       "name": "Field 3",
       "type": "number",
       "values": Array [
-        null,
+        undefined,
```
Comment (torkelo) [inline packages/grafana-data/src/utils/__snapshots__/csv.test.ts.snap]: is this correct?

Comment (ryantxu) [reply packages/grafana-data/src/utils/__snapshots__/csv.test.ts.snap]: We should choose what behavior we want, I go back and forth.

The question is what should be filled into the values that do not have anything configured.  To match the behavior of an array using an index that has nothing, I picked `undefined`, but previously I had explicitly filled up the missing values with `null`

File: packages/grafana-data/src/utils/dataFrameHelper.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-data/src/utils/dataFrameHelper.ts]: FYI, same things in on all Vector instances -- when a Vector is written to JSON, it looks like a normal Array

File: packages/grafana-data/src/utils/__snapshots__/csv.test.ts.snap (hunk 2 above)
Comment (torkelo) [reply packages/grafana-data/src/utils/__snapshots__/csv.test.ts.snap]: Think null is better, undefined is not valid json (for over the wire at least) and null is what Grafana normally use for missing data point

**Pull request merged**